### ISSUE TITLE
stock_info.tickers_sp500() returning strings of underscores

### ISSUE THREAD

**Issue by GrainsofSalt** [atreadw1492/yahoo_fin#105]
`
from yahoo_fin import stock_info as si
tickers = si.tickers_sp500()
print(tickers)
`

returns this:
['-', '-', '-', '-', '-', '-', '-', '-', '-', '-', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '--', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '---', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '----', '-----', '-----', '-----']

**Comment by 3den** [comment]
I'm having the same problem, that only to happens tickers_sp500, dow and nasdaq work fine

**Comment by 3den** [comment]
I sent a pr to fix it, not sure when it will get merged

**Comment by rnbguy-zz** [comment]
Any updates on this?

**Comment by j2r2jones** [comment]
I'm also having this issue, any updates?

**Comment by peanutsee** [comment]
A little late, but this package seems old and it's not being maintained. A work around:

```python
sp500_tickers = pd.read_html("https://en.wikipedia.org/wiki/List_of_S%26P_500_companies")[0]['Symbol'].to_list()
```

**Comment by GrainsofSalt** [comment]
Yeah, it seems the problem is with their API. The underlying methods still work. I would suggest as a work around people just use them directly. For sake of ease:

import pickler as pickle
import ftplib
import io
import pandas as pd

def tickers_snp():
    tickers = pd.read_html('https://en.wikipedia.org/wiki/List_of_S%26P_500_companies')[0].Symbol.tolist()
    return tickers

def tickers_nasdaq(include_company_data = False):
    ftp = ftplib.FTP("ftp.nasdaqtrader.com")
    ftp.login()
    ftp.cwd("SymbolDirectory")
    
    r = io.BytesIO()
    ftp.retrbinary('RETR nasdaqlisted.txt', r.write)
    
    if include_company_data:
        r.seek(0)
        data = pd.read_csv(r, sep = "|")
        return data
    
    info = r.getvalue().decode()
    splits = info.split("|")
    
    
    tickers = [x for x in splits if "\r\n" in x]
    tickers = [x.split("\r\n")[1] for x in tickers if "NASDAQ" not in x != "\r\n"]
    tickers = [ticker for ticker in tickers if "File" not in ticker]    
    
    ftp.close()    

    return tickers

def tickers_dow(include_company_data = False):
    site = "https://en.wikipedia.org/wiki/Dow_Jones_Industrial_Average"
    
    table = pd.read_html(site, attrs = {"id":"constituents"})[0]
    
    if include_company_data:
        return table

    dow_tickers = sorted(table['Symbol'].tolist())
    
    return dow_tickers    

def tickers_clean(tickers):
    return [s for s in tickers if not (len(s) == 5 and s[-1] in {'W', 'R', 'P', 'Q'})]

if __name__=="__main__":
    tickers = tickers_clean(tickers_snp())
    print("snp: ", tickers, "\n")
    pickle.pickleDump(tickers, "./tickers/snp")

    tickerss = tickers_clean(tickers_nasdaq())
    print("nasdaq: ", tickerss, "\n")
    pickle.pickleDump(tickerss, "./tickers/nasdaq")

    tickersss = tickers_clean(tickers_dow())
    print("dow: ", tickersss, "\n")
    pickle.pickleDump(tickersss, "./tickers/dow")

    print("SNP: ", pickle.pickleRead("./tickers/snp"), "\n")
    print("NASDAQ: ", pickle.pickleRead("./tickers/nasdaq"), "\n")
    print("DOW: ", pickle.pickleRead("./tickers/dow"), "\n")


This will also pickle the output to a local folder called "tickers" incase something changes with the scraping in the future, you can have the lists locally. 

-- Sorry, not sure why it is formatting poorly.
